packages/wallet/src/cip30.ts

```diff
@@ -92,9 +102,23 @@ export const createWalletApi = (
       throw new ApiError(500, 'Nope');
     }
   },
-  getCollateral: async () => {
-    logger.warn('getCollateral is not implemented');
-    return null;
+  getCollateral: async (params: { amount: Cbor }): Promise<Cbor[] | null> => {
```

Comment:
This might be a thing:
> NOTE: some wallets may be ignoring the amount parameter, in which case it might be possible to call the function without it, but this behavior is not recommended!)

```suggestion
  getCollateral: async ({ amount = cbor(MAX_COLLATERAL_AMOUNT)}: { amount?: Cbor } = {}): Promise<Cbor[] | null> => {
```

Reply:
Hm no,
- CIP specifies the amount as CBOR so we need to use `CSL.BigNum` to parse it (I guess, it's not 100% clear what type is it, could be CSL.Value as well but let's just go with BigNum). So actually this is also incorrect: `const bigIntAmount = BigInt(amount);`
- The NOTE says that it could be called without arguments, so we need a default `{}`

```
getCollateral: async ({ amount = CSL.BigNum.fromStr(MAX_COLLATERAL_AMOUNT.toString())}: { amount?: Cbor } = {}): Promise<Cbor[] | null>
```

Reply:
Since the contract expects an amount of type `string` it should be `MAX_COLLATERAL_AMOUNT.to_str() `(assuming `MAX_COLLATERAL_AMOUNT = CSL.BigNum.from_str('5000000');` ) and internally we have to map it to `BigNum` to operate.
Also, what type of string are we expecting? I know `Cbor = string` but this represents a hex string, right? So we have to build the object `CSL.BigNum.from_bytes(Buffer.from(amount,'hex'))`

Reply:
Seems right, internally you can probably do `BigInt(BigNum.toString())`